lms/djangoapps/course_api/api.py

```diff
@@ -43,11 +46,11 @@ def course_detail(request, username, course_key):
         `CourseDescriptor` object representing the requested course
     """
     user = get_effective_user(request.user, username)
-    try:
-        course = get_course_with_access(user, 'see_exists', course_key)
-    except Http404:
```

Comment:
Originally, @jcdyer was thinking that we would only return DRF exceptions.  However, I found that DRF explicitly handle's django's Http404 errors: http://www.django-rest-framework.org/api-guide/exceptions/#exception-handling-in-rest-framework-views.